lib/ansible/modules/network/aos/aos_device.py

```diff
@@ -0,0 +1,208 @@
+requirements:
+  - "aos-pyez >= 0.6.0"
+options:
+  session:
+    description:
+      - An existing AOS session as obtained by aos_login module
+    required: true
+  name:
+    description:
+      - The device serial-number; i.e. uniquely identifies the device in the
+        AOS system. Only one of I(name) or I(id) can be set.
+  id:
+    description:
+      - The AOS internal id for a device; i.e. uniquely identifies the device in the
+        AOS system. Only one of I(name) or I(id) can be set.
+  state:
+    description:
+      - Define in which state the device should be. Currently only I(normal)
+        is supported but the goal is to add I(maint) and I(decomm).
+    default: normal
+    choices: ['normal']
+  approve:
+    description:
+      - The approve argument instruct the module to convert a device in quarantine
+        mode into approved mode.
+  location:
+    description:
+      - When approving a device using the I(approve) argument, it's possible
+        define the location of the device.
+'''
+
+EXAMPLES = '''
+
+- name: Approve a new device
+  aos_device:
+    session: "{{ aos_session }}"
+    name: D2060B2F105429GDABCD123
+    state: 'normal'
+    approve: true
+    location: "rack-45, ru-18"
+'''
+
+
+RETURNS = '''
+name:
+  description: Name of the Device, usually the serial-number.
+  returned: always
+  type: str
+  sample: Server-IpAddrs
+
+id:
+  description: AOS unique ID assigned to the Device
+  returned: always
+  type: str
+  sample: fcc4ac1c-e249-4fe7-b458-2138bfb44c06
+
+value:
+  description: Value of the object as returned by the AOS Server
+  returned: always
+  type: dict
+  sample: {'...'}
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.aos import get_aos_session, check_aos_version, find_collection_item
+
+def aos_device_normal(module, aos, dev):
+
+    margs = module.params
+
+    # If approve is define, check if the device needs to be approved or not
+    if margs['approve'] is not None:
+
+        if dev.is_approved:
+            module.exit_json(changed=False,
+                             name=dev.name,
+                             id=dev.id,
+                             value=dev.value)
+
+        if not module.check_mode:
+            try:
+                dev.approve(location=margs['location'])
+            except (SessionError, SessionRqstError):
+                module.fail_json(msg="Unable to approve device")\
+
+        module.exit_json(changed=True,
+                         name=dev.name,
+                         id=dev.id,
+                         value=dev.value)
+    else:
+        # Check if the device is online
+        if dev.state in ('OOS-READY','IS-READY'):
+            module.exit_json(changed=False,
+                             name=dev.name,
+                             id=dev.id,
+                             value=dev.value)
+        else:
+            module.fail_json(msg="Device is not in 'normal' state")
```

Comment:
Would it be worth updating the message to "Device is in '%s' rather than 'normal' state"?